zamba/models/manager.py

```diff
@@ -1,127 +1,97 @@
-                return e
-
-        raise ValueError(f"Key '{name}' not in ModelName Enum.")
-
-
-class ModelName(Enum, metaclass=GetItemMeta):
-    """Allows easy control over which Model subclass to load. To add a new model class, add a line like ``NEW_MODEL
-    = ('new_model', NewModelClass)``
-
-        Args:
-            string (str) : string used to reference the model model
-            model (Model) : model class to instantiate into manager
-
-    """
-
-    WINNING = ('cnnensemble', CnnEnsemble)
-    SAMPLE = ('sample', SampleModel)
-
-    def __init__(self, string, model):
-        self.string = string
-        self.model = model
-
-    def __eq__(self, other):
-        return other == self.value
+default_train = TrainConfig()
+default_predict = PredictConfig()
 
 
 class ModelManager(object):
     """Mediates loading, configuration, and logic of model calls.
 
         Args:
-            model_path (str | Path) : path to model weights and architecture
-                Required argument. Will be instantiated as Model object.
-            proba_threshold (float) : probability threshold for classification
-                Defaults to ``None``, in which case class probabilities are returned.
-            tempdir (str | Path) : path to temporary directory
-                If specific temporary directory is to be used, its path is passed here. Defaults to ``None``.
-            verbose (bool) : controls verbosity of prediction, training, and tuning methods
-                Defaults to ``True`` in which case training, tuning or prediction progress will be logged.
-            model_class (str) : controls whether sample model class or production model class is used
-                Defaults to "winning". Must be "winning" or "sample".
+            train_config (TrainConfig) : Configuration for model training.
+            predict_config (PredictConfig): Configuration for inference.
     """
     def __init__(self,
-                 model_path=Path('.'),
-                 proba_threshold=None,
-                 output_class_names=False,
-                 tempdir=None,
-                 verbose=False,
-                 model_class='cnnensemble',
-                 model_kwargs=dict()):
+                 train_config=default_train,
+                 predict_config=default_predict):
 
+        self.train_config = train_config
+        self.predict_config = predict_config
         self.logger = logging.getLogger(f"{__file__}")
 
-        self.model_path = Path(model_path)
-        self.model_class = ModelName[model_class].model
-
-        self.tempdir = tempdir
-        self.model = self.model_class(model_path, verbose=verbose, **model_kwargs)
-        self.proba_threshold = proba_threshold
-
-        self.output_class_names = output_class_names
+    @staticmethod
+    def from_config(config):
+        if not isinstance(config, ModelManager):
+            config = ModelConfig.parse_file(config)
+        return ModelManager(**dict(config))
```

Comment:
We want `config.dict()` instead

Reply:
hmm don't think so. IIRC this was from my learning documented here: https://drivendata.slack.com/archives/CUB01MUBT/p1617745763040200

if you pass as `config.dict()`, then you get `ModelManager(train_config={n_epochs:...}` vs if you pass `dict(config`, then you get `ModelManager(train_config=TrainConfig(n_epochs=...))` -- we want the later in order to do `self.train_config.n_epochs` in all cases rather than `self.train_config['n_epochs']` if we passed in a config file

Reply:
Hm, interesting. To me that's a signal that we're not organized quite right then. Nested Pydantic models should be serialized/deserialized at the same time. To me it's something to be wary of to depend on those classes _not_ being serialized to dicts because `dict(BaseModel)` doesn't interpret the nesting correctly.